include/eve/algo/concepts/eve_iterator.hpp

```diff
@@ -55,6 +57,7 @@ namespace eve::algo
     std::totally_ordered_with<I, unaligned_t<I>> &&
     std::totally_ordered_with<I, partially_aligned_t<I>> &&
     requires(I i, std::ptrdiff_t n) {
+       { std::declval<typename I::value_type>() };
```

Comment:
you don't need the declval.
`typename I::value_type;` is enough.
See Type Requirements at https://en.cppreference.com/w/cpp/language/constraints